File: src/main/java/com/nibado/projects/advent/y2016/AdventOfCode.java
package com.nibado.projects.advent.y2016;

import java.util.Arrays;
import java.util.List;

public class AdventOfCode {
    public static void main(String... argv) {
        List<Runnable> solutions = Arrays.asList(
                new Day01(),
                new Day02());

        solutions.forEach(Runnable::run);
    }
}

File: src/main/kotlin/com/nibado/projects/advent/y2016/AdventOfCode.kt
package com.nibado.projects.advent.y2016

import com.nibado.projects.advent.Reflect
import com.nibado.projects.advent.Runner

fun main(args: Array<String>) {
    val days = Reflect.getDays(2016)

    val day = if(args.isEmpty()) 0 else args[0].toInt()

    Runner.run(2016, days, day)
}